src/Nest/CommonOptions/TimeUnit/Time.cs

```diff
@@ -16,57 +16,99 @@ public class Time : IComparable<Time>, IEquatable<Time>
 		private const double MillisecondsInAnHour = MillisecondsInAMinute * 60;
 		private const double MillisecondsInAMinute = MillisecondsInASecond * 60;
 		private const double MillisecondsInASecond = 1000;
+		private const double NanosecondsInAMillisecond = 100;
+		private const double MicrosecondsInAMillisecond = 10;
 
-		private static readonly Regex _expressionRegex = new Regex(@"^(?<factor>[-+]?\d+(?:\.\d+)?)(?<interval>(?:y|M|w|d|h|m|s|ms))?$", RegexOptions.Compiled | RegexOptions.ExplicitCapture);
+		private static readonly Regex ExpressionRegex =
+			new Regex(@"^(?<factor>[-+]?\d+(?:\.\d+)?)\s*(?<interval>(?:y|w|d|h|m|s|ms|nanos|micros))?$",
+				RegexOptions.Compiled | RegexOptions.ExplicitCapture | RegexOptions.IgnoreCase);
```

Comment:
I would keep this a case-sensitive match and include `M` for Months, along with the parsing `ToEnum<TimeUnit>()` as case-sensitive; this better reflects what's accepted by Elasticsearch